include/libopencm3/stm32/f4/rcc.h

```diff
@@ -499,17 +559,45 @@
 #define RCC_PLLSAICFGR_PLLSAIQ_SHIFT		24
 #define RCC_PLLSAICFGR_PLLSAIQ_MASK		0xF
 
+/* RCC_PLLSAICFGR[18:16]: PLLSAIP */
+#define RCC_PLLSAICFGR_PLLSAIP_SHIFT		16
+#define RCC_PLLSAICFGR_PLLSAIP_MASK		0x3
+
 /* RCC_PLLSAICFGR[14:6]: PLLSAIN */
-#define RCC_PLLSAICFGR_PLLSAIN_SHIFT		14
+#define RCC_PLLSAICFGR_PLLSAIN_SHIFT		6	
 #define RCC_PLLSAICFGR_PLLSAIN_MASK		0x1FF
 
 
 /* --- RCC_DCKCFGR values -------------------------------------------------- */
-#define RCC_DCKCFGR_PLLSAIDIVR_MSK                 (0x3 << 16)
-#define RCC_DCKCFGR_PLLSAIDIVR_DIVR_2              (0x0)
-#define RCC_DCKCFGR_PLLSAIDIVR_DIVR_4              (0x1)
-#define RCC_DCKCFGR_PLLSAIDIVR_DIVR_8              (0x2)
-#define RCC_DCKCFGR_PLLSAIDIVR_DIVR_16             (0x3)
+#define RCC_DCKCFGR_DSISEL			(1 << 29)
+#define RCC_DCKCFGR_SDMMCSEL			(1 << 28)
+#define RCC_DCKCFGR_48MSEL			(1 << 27)
+#define RCC_DCKCFGR_TIMPRE			(1 << 24)
+
+#define RCC_DCKCFGR_SAI1BSRC_SHIFT		22
+#define RCC_DCKCFGR_SAI1BSRC_MASK		0x3
+
+#define RCC_DCKCFGR_SAI1ASRC_SHIFT		20
+#define RCC_DCKCFGR_SAI1ASRC_MASK		0x3
+
+/* Values for the BSRC and ASRC fields */
+#define RCC_DCKCFGR_SAI1SRC_SAIQ		0x0
+#define RCC_DCKCFGR_SAI1SRC_I2SQ		0x1
+#define RCC_DCKCFGR_SAI1SRC_ALT			0x2
+#define RCC_DCKCFGR_SAI1SRC_ERROR		0x3
+
+#define RCC_DCKCFGR_PLLSAIDIVQ_SHIFT		8
+#define RCC_DCKCFGR_PLLSAIDIVQ_MASK		0x1f
+
+#define RCC_DCKCFGR_PLLI2SDIVQ_SHIFT		0
+#define RCC_DCKCFGR_PLLI2SDIVQ_MASK		0x1f
+
+#define RCC_DCKCFGR_PLLSAIDIVR_SHIFT		16
+#define RCC_DCKCFGR_PLLSAIDIVR_MASK		0x3
+#define RCC_DCKCFGR_PLLSAIDIVR_DIVR_2		0x0
+#define RCC_DCKCFGR_PLLSAIDIVR_DIVR_4		0x1
+#define RCC_DCKCFGR_PLLSAIDIVR_DIVR_8		0x2
+#define RCC_DCKCFGR_PLLSAIDIVR_DIVR_16		0x3
```

Comment:
I'll probably move this back up above PLLSAIDIVQ to keep the bits in order if that's ok with you